benchmarks/dynamo/genai_layers/utils.py

```diff
@@ -0,0 +1,241 @@
+import os
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import Any, Callable, Optional
+
+import matplotlib.pyplot as plt
+
+import torch
+from torch._inductor.runtime.benchmarking import benchmarker
+
+
+def benchmark_kernel_in_milliseconds(func: Callable, *args, **kwargs) -> float:
+    # warmup
+    for _ in range(5):
+        func(*args, **kwargs)
+    return benchmarker.benchmark_gpu(lambda: func(*args, **kwargs))
```

Comment:
nit: we should probably enforce no recompiles after warm-up?

for _ in range(5):
    func(*args, **kwargs)
with torch.compiler.set_stance("fail_on_recompile"):
    return benchmarker.benchmark_gpu(lambda: func(*args, **kwargs))